app/routes/overview/components/LatestReadme.tsx

```diff
@@ -1,69 +1,51 @@
-import { Component } from 'react';
+import { useEffect, useState } from 'react';
+import { Collapse } from 'react-collapse';
 import Card from 'app/components/Card';
 import Icon from 'app/components/Icon';
 import { Image } from 'app/components/Image';
 import { Flex } from 'app/components/Layout';
 import { readmeIfy } from 'app/components/ReadmeLogo';
+import type { Readme } from 'app/models';
 import styles from './LatestReadme.css';
 
 type Props = {
-  expanded: boolean;
-  readmes: Array<Record<string, any>>;
+  expandedInitially: boolean;
+  readmes: Array<Readme>;
 };
-type State = {
-  expanded: boolean;
-};
-
-class LatestReadme extends Component<Props, State> {
-  state = {
-    expanded: this.props.expanded,
-  };
-
-  // eslint-disable-next-line
-  componentWillReceiveProps(nextProps: Props) {
-    if (nextProps.expanded !== this.state.expanded) {
-      this.setState({
-        expanded: nextProps.expanded || false,
-      });
-    }
-  }
-
-  render() {
-    const { expanded } = this.state;
 
-    const toggle = () =>
-      this.setState((state) => ({
-        expanded: !state.expanded,
-      }));
-
-    const { readmes = [] } = this.props;
-    return (
-      <Card className={styles.latestReadme}>
-        <button className={styles.heading} onClick={toggle}>
-          <Flex justifyContent="space-between" alignItems="center">
-            {readmeIfy('readme')}
-            <Icon name={expanded ? 'close' : 'arrow-down'} />
-          </Flex>
-        </button>
-
-        {expanded && (
-          <Flex
-            wrap
-            justifyContent="space-around"
-            style={{
-              paddingTop: 20,
-            }}
-          >
-            {readmes.slice(0, 4).map(({ image, pdf, title }) => (
-              <a key={title} href={pdf} className={styles.thumb}>
-                <Image src={image} alt={`Cover of ${title}`} />
-              </a>
-            ))}
-          </Flex>
-        )}
-      </Card>
-    );
-  }
-}
+const LatestReadme = ({ readmes, expandedInitially }: Props) => {
+  const [expanded, setExpanded] = useState(expandedInitially);
+
+  useEffect(() => {
+    setExpanded(expandedInitially);
+  }, [expandedInitially]);
```

Comment:
Do you need this `useEffect`? In my mind it has no effect :thinking:

Reply:
In prod, it will not have any effect, since all the fetching is done server side before it reaches the client. However, when running this locally, the value changes as the fetching is happening. And for some reason, the component does not re render when the initial value changes as the fetches happen. Therefore, the initial value ends up wrong, if you do not include this.